example-app/src/main/java/com/adyen/checkout/example/ui/googlepay/compose/SessionsGooglePayScreen.kt

```diff
@@ -71,26 +69,39 @@ internal fun SessionsGooglePayScreen(
             )
         },
     ) { innerPadding ->
-        val googlePayState by viewModel.googlePayState.collectAsState()
+
+        when (eventsState) {
+            is SessionsGooglePayEvents.ComponentData -> {
+                googlePayComponent = getGooglePayComponent(componentData = eventsState.data)
+            }
+            is SessionsGooglePayEvents.Action -> {
+                LaunchedEffect(eventsState.action) {
+                    googlePayComponent.handleAction(eventsState.action, activity)
+                }
+            }
+            is SessionsGooglePayEvents.Intent -> {
+                LaunchedEffect(eventsState.intent) {
+                    googlePayComponent.handleIntent(eventsState.intent)
+                }
+            }
+        }
+
         SessionsGooglePayContent(
+            googlePayComponent = googlePayComponent,
```

Comment:
Btw I just tested the branch locally and this `lateinit` field is causing a crash here (and a few other places) when the screen is being created:
```
kotlin.UninitializedPropertyAccessException: lateinit property googlePayComponent has not been initialized
```

I'm not sure anymore if this solution will work. I'll try to find a solution but let me know if you have any suggestions.

Reply:
I didn't have a client key implementation readily available, and haven't tested it. Let me take a look tomorrow. What's the quickest way to have a client key added?

Reply:
Yeah sorry my bad, I should have tested it myself before merging but I didn't want to keep stalling this PR so I thought I'll fix any errors after merging 😅 

If you have an Adyen test account already then you can just follow [this step in the integration guide](https://docs.adyen.com/development-resources/client-side-authentication/#get-your-client-key). Otherwise don't worry about it, I'll try to figure it out.